### PR TITLE
T23208: Drop libwobbly dependency, switch to libwobbly-glib-0.3

### PR DESCRIPTION
**Pull request by smspillaz**
This PR drops the dependency on libwobbly's C++ API and moves to using libwobbly-glib (the C API). Amongst other things, using this API means that we no longer have a single C++ file cluttering up the build.

https://phabricator.endlessm.com/T23208

### CODE REVIEW

**Comment by joaquimrocha** [comment]
Jenkins: retest this please.

**Comment by smspillaz** [comment]
Ah, we'll need to include libwobbly-glib-0.3 in the distro . I'll chat to @cosimoc about that, the easiest way is probably to migrate it to `endlessm` and create a jenkins job

**Comment by cosimoc** [comment]
Update: I created the repository now in endlessm as a fork of the original smspillaz one, and added the libwobbly job to the eos-build template. I can't run the jenkins-jobs command to update it at the moment. I will try again in a bit.

File: meson.build [hunk 1]
```diff
@@ -102,6 +102,10 @@ if enable_recorder
   recorder_deps += [gst_dep, gst_base_dep, gtk_dep, x11_dep]
 endif
 
+cc = meson.get_compiler('c')
+
+m_dep = cc.find_library('m', required: false)
```
Comment (cosimoc) [inline meson.build]: Why is this required? I'd avoid diverging from the upstream build file if possible.

File: src/wobbly-effect.c [hunk 2]
```diff
@@ -298,12 +297,11 @@ endless_shell_fx_wobbly_ungrab (EndlessShellFXWobbly *effect)
    * of the animation */
   if (priv->timeout_id)
     {
```
Comment (cosimoc) [inline src/wobbly-effect.c]: Remove curly braces here and below, now that they only have one line

File: meson.build (hunk 1 above)
Comment (smspillaz) [reply meson.build]: Its required since EndlesShellFX depends on `libm`. I can probably get rid of EndlessShellFX now that we dont need to compile it separately but would rather do that in another story since it's a lot of churn.
